packages/runtime/datastore-definitions/src/dataStoreRuntime.ts

```diff
@@ -88,6 +88,15 @@ export interface IFluidDataStoreRuntime
 	 */
 	createChannel(id: string | undefined, type: string): IChannel;
 
+	/**
+	 * This api allows adding channel to data store after it was created.
+	 * The channel type should be present in the registry, otherwise the runtime would reject
+	 * the channel. The runtime used to create the channel object should be same to which
+	 * it is added.
+	 * @param channel - channel which needs to be added to the runtime.
+	 */
+	addChannel(channel: IChannel): void;
```

Comment:
Why does this need to be exposed? createChannel is there for channel creation. This seems like bindToContext equivalent which we wouldn't want to expose?

Reply:
added you to teams thread, but also added really large comment on the interface definition.